inspirehep/modules/migrator/tasks.py

```diff
@@ -224,27 +215,28 @@ def continuous_migration(skip_files=None):
         )
     redis_url = current_app.config.get('CACHE_REDIS_URL')
     r = StrictRedis.from_url(redis_url)
-    lock = Lock(r, 'continuous_migration', expire=120, auto_renewal=True)
-    if lock.acquire(blocking=False):
-        try:
+    try:
+        with distributed_lock('continuous_migration'):
+            if not r.llen('legacy_records'):
+                LOGGER.info("Continuous_migration queue is empty, nothing to do.")
+                return
+
             while r.llen('legacy_records'):
-                raw_record = r.lrange('legacy_records', 0, 0)
-                if raw_record:
-                    migrate_and_insert_record(
-                        zlib.decompress(raw_record[0]),
-                        skip_files=skip_files,
-                    )
-                    db.session.commit()
-                r.lpop('legacy_records')
-        finally:
-            lock.release()
-    else:
+                raw_records = (r.lpop('legacy_records') for _ in range(CHUNK_SIZE))
+                decompressed_records = [
+                    zlib.decompress(raw_record) for raw_record in raw_records if raw_record
+                ]
+                insert_into_mirror(decompressed_records)
+            migrate_from_mirror(wait_for_results=True, skip_files=skip_files)
+    except DistributedLockError:
         LOGGER.info("Continuous_migration already executed. Skipping.")
 
 
-@shared_task(ignore_result=False, queue='migrator')
+@shared_task(bind=True, queue='migrator')
 @disable_orcid_push
```

Comment:
this is troublesome, as we don't want to disable the orcid push in the continuous migration. I will changed it.